src/plugin.js

```diff
@@ -309,9 +324,17 @@ const plugin = function(options) {
   // because it doesn't make sense to pass it to each Overlay component.
   delete settings.overlays;
 
-  this.overlays_ = overlays.map(
-    o => this.addChild('overlay', videojs.mergeOptions(settings, o))
-  );
+  this.overlays_ = overlays.map(o => {
+    let mergeOptions = videojs.mergeOptions(settings, o);
+
+    // Attach bottom aligned overlays to the control bar so
+    // they will adjust positioning when the control bar minimizes
+    if (this.controlBar && mergeOptions.align.indexOf('bottom') !== -1) {
+      return this.controlBar.addChild('overlay', mergeOptions);
+    }
+
+    return this.addChild('overlay', mergeOptions);
```

Comment:
We have the "early-exit" rule where since in the `if` above we exit with the `return`, we don't need an else.